packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.test.ts

```diff
@@ -0,0 +1,295 @@
+      setFacetsState(
+        buildMockCommerceRegularFacetResponse({
+          facetId,
+          values: [
+            {value: 'Acme', state: 'idle'},
+            breadcrumb,
+          ] as RegularFacetValue[],
+        })
+      );
+    });
+
+    it('generates breadcrumbs', () => {
+      expectBreadcrumbToBePresentInState(breadcrumb);
+    });
+
+    describe.each([
+      ['selected', toggleSelectFacetValue],
+      ['excluded', toggleExcludeFacetValue],
+    ])('#deselect when facet is %s', generateDeselectionTestCases(breadcrumb));
+  });
+
+  describe('numeric facet breadcrumbs', () => {
+    const breadcrumb = {
+      start: 100,
+      end: 200,
+      state: 'selected',
+    } as NumericFacetValue;
+
+    beforeEach(() => {
+      setFacetsState(
+        buildMockCommerceNumericFacetResponse({
+          facetId,
+          values: [
+            {start: 0, end: 100, state: 'idle'},
+            breadcrumb,
+          ] as NumericFacetValue[],
+        })
+      );
+    });
+
+    it('generates breadcrumbs', () => {
+      expectBreadcrumbToBePresentInState(breadcrumb);
+    });
+
+    describe.each([
+      ['selected', toggleSelectNumericFacetValue],
+      ['excluded', toggleExcludeNumericFacetValue],
+    ])('#deselect when facet is %s', generateDeselectionTestCases(breadcrumb));
+  });
+
+  describe('date facet breadcrumbs', () => {
+    const breadcrumb = {
+      start: '2020',
+      end: '2021',
+      state: 'selected',
+    } as DateFacetValue;
+
+    beforeEach(() => {
+      setFacetsState(
+        buildMockCommerceDateFacetResponse({
+          facetId,
+          values: [
+            {start: '2019', end: '2020', state: 'idle'},
+            breadcrumb,
+          ] as DateFacetValue[],
+        })
+      );
+    });
+
+    it('generates breadcrumbs', () => {
+      expectBreadcrumbToBePresentInState(breadcrumb);
+    });
+
+    describe.each([
+      ['selected', toggleSelectDateFacetValue],
+      ['excluded', toggleExcludeDateFacetValue],
+    ])('#deselect when facet is %s', generateDeselectionTestCases(breadcrumb));
+  });
+
+  describe('category facet breadcrumbs', () => {
+    const breadcrumb = {
+      value: 'Shoes',
+      path: ['Shoes'],
+      children: [],
+      state: 'selected',
+    } as unknown as CategoryFacetValue;
+
+    beforeEach(() => {
+      setFacetsState(
+        buildMockCategoryFacetResponse({
+          facetId,
+          values: [
+            {
+              value: 'Hats',
+              path: ['Hats'],
+              children: [],
+              state: 'idle',
+              moreValuesAvailable: false,
+              numberOfResults: 10,
+              isLeafValue: true,
+            },
+            breadcrumb,
+          ] as CategoryFacetValue[],
+        })
+      );
+    });
+
+    it('generates breadcrumbs', () => {
+      expectBreadcrumbToBePresentInState(breadcrumb);
+    });
+
+    describe.each([['selected', toggleSelectCategoryFacetValue]])(
```

Comment:
No need for a describe.each in this case since there's just one action (selected) to test